src/access/Ownable2Step.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import {IOwnable2Step} from "src/interfaces/access/IOwnable2Step.sol";
+
+import {Ownable} from "src/access/Ownable.sol";
+
+/// @notice Gas-optimized Ownable2Step helpers.
+/// @dev Reference: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/access/Ownable2Step.sol
+contract Ownable2Step is IOwnable2Step, Ownable {
+    address private _pendingOwner;
+
+    /// @dev Initializes the contract setting the deployer as the initial owner.
+    constructor(address initialOwner) Ownable(initialOwner) {}
+
+    /* PUBLIC */
+
+    /// @inheritdoc IOwnable2Step
+    function pendingOwner() public view virtual returns (address) {
+        return _pendingOwner;
+    }
+
+    /// @inheritdoc IOwnable2Step
+    function acceptOwnership() public virtual {
+        address sender = msg.sender;
```

Comment:
No need to cache the sender because msg.sender is actually cheaper than using memory.